### DISCUSSION TITLE
[BP#89] Utiliser des serveurs virtualisés

### DISCUSSION THREAD

**Discussion by ddebarbieux** [cnumr/best-practices#154] [Bonnes pratiques] [answered]
Utiliser la virtualisation est efficace si :
- Les environnements de test, de recette… sont décommissionnés lors qu'ils ne sont plus actifs.
- L'équipe infra est une gestion raisonnée de ses ressources.

Je propose donc d'ajouter ces éléments à la description de la bonne pratique.

**Comment by brpasi** [comment]
Hello. Je comprends pas bien le 2nd point :
- L'équipe infra est une gestion raisonnée de ses ressources.

Peux-tu préciser stp ?

**Comment by devcrafting** [comment]
On pourrait aussi ajouter la containairisation et citer Docker.
On peut aussi diminuer le pourcentage de la règle de validation

**Comment by arnolem** [comment]
Le sujet est vaste mais utile. Les questions que l'on peut se poser sur le sujet sont nombreuses au niveau du serveur (au delà du datacenter qui est un autre sujet)
 - Est-ce mieux d'avoir un serveur dédié (bien dimensionné) ou un serveur virtuel (qui gâche des ressources pour l'émulation mais qui permet plus de flexibilité) ?
  - Vaut-il mieux une VM ou un container type LXC ?
  - Vaut-il mieux virtualisé un OS complet (Exemple courant avec LXC) ou containériser des services (Courant sur Docker) ?
  - Vaut-il mieux des SSD ou des HDD ?
  - Vaut-il mieux pleins de "petits" serveurs ou moins de serveurs mais plus "gros" ?
  - Vaut-il mieux aller vers du Cloud (Tarif à la consommation, mutualisation des ressources ... Mais infrastructure distante, parfois sur-dimensionnée, ...) ?
  - On parle des "Cpanel and co"  VS "Installer soit même un serveur light"

J'ai déjà ma petite idée sur pas mal d’éléments mais avant d'arriver à "Utiliser des serveurs virtualisés", il faut discuter je pense.

**Comment by ACTLEM** [comment]
Plutôt pour une alerte sur le fait de bien décommissionner quand ce n'est pas utilisé. Pourquoi pas ajouter une BP sur l'élasticité en général.

@LGnord, c'est possible de créer un point visio (15-30mins) pour discuter avec les personnes qui seraient intéressées afin de proposer concrètement les modifications, ajouts svp?

**Reply by devcrafting** [reply]
Intéressé aussi

**Reply by ddebarbieux** [reply]
@ACTLEM @devcrafting 
Quel créneau vous convient le mieux : demain (mardi) vers 10h ou mercredi vers 9h

**Reply by ACTLEM** [reply]
Perso, je suis full demain et mercredi. Désolé.

**Reply by ddebarbieux** [reply]
Quelques points pour éclaisir ma pensé : 
- Utiliser des serveurs virtualisés : aujourd'hui, tout le monde le fait (c'est mon ressenti). Il faut donc compléter un peu la bonne pratique.
- Aucune des 115 bonnes pratiques ne parlent de la fin de vie. C'est un objectif de la nouvelle version (si j'ai bien compris). Parler de décommissionnement me semble donc judicieux.
- Choisir un bon herbergeur est un sujet complexe qui dépasse problement le livre (même si la bonne pratique sur l'herbergeur vert y fait référence).
- Je suis ni obs ni même devOps. Je ne suis pas capable de redigger seul cette BP.

@arnolem, tu semble avoir aussi des éléments de réponses.
@ACTLEM, tu as des propositions de dates ?

**Reply by ACTLEM** [reply]
En phase sur la notion de décomissionnement. L'idée d'utiliser des serveurs virtualisés est de mutualiser et réduire (en théorie) le nombre d'équipements fabriqués et alimentés, mais je pense que tout le monde l'a en tête. De toute façon les cloud provider ont intérêt pour des raisons économiques de réduire leur facture d'électricité donc réduire leurs équipements et d'optimiser au mieux à besoin équivalent. Petit retour terrain: les entreprises qui ont des gros leviers côté datacenter (suite à des ACV) sont celles qui ont leur propre datacenter...mais sous réserve qu'elles aient une vision raisonnée ^^
Dispo uniquement à l'heure du midi ou parfois en soirée. Je pense que si @devcrafting et @arnolem sont présents, ma présence n'est pas forcément nécessaire.

**Reply by devcrafting** [reply]
Dispo pour 30min mercredi 9h

**Reply by ddebarbieux** [reply]
@devcrafting @arnolem 
Rendez-vous à 9h00
https://global.gotomeeting.com/join/653132581

L'échange est bien entendu ouvert à tous !

**Reply by devcrafting** [reply]
pour la partie décommissionnement, c'est pas mal traité dans https://github.com/cnumr/best-practices/blob/main/chapters/BP_085_fr.md (@LGnord va ouvrir une discussion dessus)

on est d'accord avec @LGnord sur le fait que la virtualisation/contairisation est très courante aujourd'hui donc cette BP pourrait plutôt évoluer vers les choix d'infrastructure : la renommer et aiguiller avec toutes les questions de @arnolem (qui n'a pas dû pouvoir se libérer). @LGnord et moi-même on est un peu "light" sur la partie Ops (plus utilisateur) => **nouveau point avec @arnolem et @ACTLEM semaine prochaine ?**

**Reply by tbroyer** [reply]
À ce sujet, je viens d'apprendre que Scaleway a lancé Serverless Functions et Serverless Containers, quelqu'un a des retours sur ces approches ? (du coup essentiellement Serverless Functions / Amazon Lambda / Google Cloud Functions)
On tendrait à faire du micro-service 😠, mais chaque service n'est démarré qu'à la demande et décommissionné quasiment dans la foulée ☺️

**Reply by arnolem** [reply]
Désolé, je viens juste de voir les échanges et je suis non libre ce matin.
Je vais essayer de suivre vos prochains échanges

**Reply by ACTLEM** [reply]
Petit exemple sur le servless, une entreprise de boîte de conserves connue (vous avez une chance sur deux mais indice elle est dans le Nord ^^) a migré beaucoup de ses services sur du Amazon Lambda. L'intérêt a surtout été financier et cela a été confirmé donc indirectement moins de ressources utilisées (donc environnementale). Il s'agit surtout de leurs services backend (et ils ont très peu de sites web). Donc oui le servless a un intérêt, par contre pour le web les use case sont moindres, cela va surtout être pour les services asynchrones et stateless, et j'avoue que les seules exemples que j'ai sont des services d'alimentation d'un catalogue produit pour un site e-commerce (dont très très back et sûrement hors scope du guide)...à voir pour les services de notifications (mais pas de retour de terrain perso).

**Reply by ACTLEM** [reply]
Des nouvelles par rapport à cette BP? On crée une issue et @LGnord et @devcrafting vous proposez quelque chose?

**Comment by rlemaire** [comment]
Je suis pour parler de mettre en place une architecture élastique qui s'adapte à la charge.
Je pense que c'est un sujet potentiel de nouvelle BP. je peux prendre l'issue si besoin.

**Reply by ACTLEM** [reply]
Sur le sujet de l'élasticité, pour moi cela reste dans le cadre de cette BP, non?

**Reply by rlemaire** [reply]
C'est toi qui disait : « Plutôt pour une alerte sur le fait de bien décommissionner quand ce n'est pas utilisé. Pourquoi pas ajouter une BP sur l'élasticité en général.» :-)

On peut les mettre ensemble oui, mais virtualisé n'implique pas une architecture élastique, et la virtualisation implique d'avoir une mutualisation des machines donc c'est déjà (un peu) intéressant même avec des vms zombies, donc il faudra bien le préciser dans le titre et le texte.

Comme : 
« Utiliser des serveurs virtualisés et décommissionner ceux qui sont inactif »

**Reply by ACTLEM** [reply]
:-D, c'est vrai, c'est moi qui avais émis l'idée...comme quoi 15j après, je ne suis plus tout à fait en phase avec moi-même ^^. 
Je me suis fait la réflexion que pour avoir une archi élastique, les serveurs virtuels étaient un pré-requis.
Je te rejoins sur toute ta remarque.

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé deux issues:

- Amélioration de la BP existante pour parler de conteneurisation, améliorer la description: https://github.com/cnumr/best-practices/issues/312
- Nouvelle BP sur la mise en place d'une infrastructure élastique: https://github.com/cnumr/best-practices/issues/313
